evennia/contrib/slots_handler/readme.md

```diff
@@ -0,0 +1,110 @@
+
+**Slot category:** SlotsHandler interprets the top-level keys of a dict, or the values of a list, to be categories. Categories are important as they factor into the storage name of the slot list, so if you don't want to separate your slots, give them a very general category such as `"slots"`.
+
+**Numbered vs named slots:** Slot keys must be integers or strings. Numbered slots are anonymous and mostly will not be referred to directly. Named slots are for cases where you want to refer to them directly.
+
+**Nesting:** The system is designed with simplicity in mind. You should not try to force individual slot attributes to be more than a basic `dict`. However, since we're storing objects in these slots, it's perfectly possible to store one slot attribute inside a slot in a different attribute. Your code just has to know to look for that, because `SlotsHandler` does not. The individual building blocks are simple, but they may be assembled in more complicated ways than they can achieve on their own.
+
+### How do I use it?
+The first way is just to record how many of a thing a character has. One object can be stored in multiple slots for systems where one mechanic might fill more than one requirement. The handler doesn't support querying for specific slots, because I feel like `.all()` is sufficient for those purposes. To retrieve a specific slot, it's easy enough to use syntax like the following:
+```python
+spells = obj.slots.all()['spells']
+weapon = obj.slots.all()['equipped']['weapon']
+```
+
+### What's next?
+While I consider `SlotsHandler` complete as-is, there is room for sample commands that interact with the handler and a set of classes that a new user can plug into their game to instantly see how it works without altering their own typeclasses.
+
+## SlottedObject.slots.add
+```
+add(self, slots)
+```
+> Create an array of slots, or add additional slots to an existing array.
+
+**Args:**
+* **slots (dict):** A dict of slots to add. Since you can't add empty categories, it would be pointless to pass a list to this function, and so it doesn't accept lists for input.
+
+## SlottedObject.slots.delete
+```
+delete(self, slots)
+```
+> Delete the named category or slots.
+> WARNING: If you have anything attached in slots when they are removed, the slots' contents will also be removed. This function will return a dict of any removed slots and their contents, so it can act as a pop(), but if you don't catch that data, it WILL be lost.
+
+**Args:**
+* **slots (list or dict):** Slot categories or individual slots to delete.
+
+## SlottedObject.slots.attach
+```
+attach(self, target, slots=None)
```

Comment:
Reading this as a new potential user I don't understand what "attach the target in all slots" means. Why would you want to regularly add the target object to *all* slots? Could you elaborate?

Reply:
I also think most of these could probably do with a brief example (not strictly necessary but helpful)

Reply:
> Reading this as a new potential user I don't understand what "attach the target in all slots" means.

The full clause is "attach the target in all slots it consumes", but that's probably not the best way to word it. I'll make sure to talk about it more thoroughly in the walkthrough. The reason for the "all slots" wording is the fact that the `attach()` method will only work if every one of the slots requested is available.